### PR TITLE
SIP-378 Section C: Fix maker/taker set to zero bug

### PR DESCRIPTION
**Pull request by leomassazza**
There's an issue where maker or taker fees are set to zero that expresses when attempting to send 0 fees to a referrer. The fix just adds a conditional on zero fees collected.

### CODE REVIEW

**Review by kaleb-keny — APPROVED** [review]
simplest fix ever

File: markets/perps-market/contracts/storage/GlobalPerpsMarketConfiguration.sol [hunk 1]
```diff
@@ -207,7 +207,7 @@ library GlobalPerpsMarketConfiguration {
         address referrer,
         PerpsMarketFactory.Data storage factory
     ) private returns (uint256 referralFeesSent) {
-        if (referrer == address(0)) {
+        if (fees == 0 || referrer == address(0)) {
```
Comment (sunnyvempati) [inline markets/perps-market/contracts/storage/GlobalPerpsMarketConfiguration.sol:210]: nit: feel like this check should exist at the caller level.  in `collectFees()` we should return (0,0) if the orderFees == 0.  this function shouldn't even be called if there's no fees to collect.

**Review by sunnyvempati — APPROVED** [review]
small nit but lgtm

**Comment by sunnyvempati** [comment]
@leomassazza i think rebase will fix the test

**Comment by leomassazza** [comment]
Closed and commits cherrypicked to main SIP-378 branch. All SIP-378 is now only one PR

**Pull request closed without merging**